src/fc/qemu/agent.py

```diff
@@ -1051,7 +1051,9 @@ def ensure(self):
                         # We didn't get the lock, so someone else is
                         # already around who will pick up the (additional)
                         # changed config later.
-                        break
+                        # This locking code is an explicit signal to potential
+                        # callers, so it must be exposed to the outside!
```

Comment:
we're not logging the exit code in the main function, so a bit of logging here would be helpful